data_structures/rtree.cpp

```diff
@@ -111,23 +127,131 @@ bool spatial::Rtree<T>::Node::insert(index_t const point_idx) {
 template<typename T>
 void spatial::Rtree<T>::Node::split(int const branch_idx) {
     // pop the overflowing branch from the 'entries' vector
+    auto const overflowing_node = entries[branch_idx].get_node();
+    entries.erase(entries.begin() + branch_idx);
 
-    // pick some seed entries using the split heuristic
+    // make some seed entries using the split heuristic
+    pick_seeds(overflowing_node->entries);
```

Comment:
I would consider returning a pair of iterators/pointers from `pick_seeds()` so that the function only has one [single responsibility](https://lispcast.com/single-responsibility-principle-for-functional-programming/). Currently it both picks the seeds and creates/inserts the new entries.